src/Spatial.Tests/Euclidean/PlaneTest.cs

```diff
@@ -80,6 +85,77 @@ public void ProjectPointOn(string ps, string rootPoint, string unitVector, strin
+            Assert.That(aPlane.SignedDistanceTo(ps[2]), Is.EqualTo(0).Within(tolerance), "ps[2]");
+        }
+
+        [TestCase(ZeroPoint, X, Y, XY, Z, "0,0,+1")] //all 4 points fall on the plane z=+1
+        [TestCase(ZeroPoint, X, Y, XY, Z, "0,0,-1")] //all 4 points fall on the plane z=-1
+        [TestCase(ZeroPoint, Y, Z, YZ, X, "+1,0,0")] //all 4 points fall on the plane x=+1
+        [TestCase(ZeroPoint, Y, Z, YZ, X, "-1,0,0")] //all 4 points fall on the plane x=-1
+        [TestCase(ZeroPoint, Z, X, ZX, Y, "0,+1,0")] //all 4 points fall on the plane y=+1
+        [TestCase(ZeroPoint, Z, X, ZX, Y, "0,-1,0")] //all 4 points fall on the plane y=-1
+        public void CreateFittedPlaneFrom_With4PointsOnAxisAlignedPlane(string sV1, string sV2, string sV3, string sV4, string sENormal, string sRootPoint)
+        {
+            var rootPoint = Point3D.Parse(sRootPoint);
+            var ps = new[]
+            {
+                Vector3D.Parse(sV1),
+                Vector3D.Parse(sV2),
+                Vector3D.Parse(sV3),
+                Vector3D.Parse(sV4),
+            }.Select(deltaV => rootPoint + deltaV);
+            var actual = Plane.CreateFittedPlaneFrom(ps);
+
+            var eNormal = UnitVector3D.Parse(sENormal);
+            var eRootPoint = Point3D.Parse(sRootPoint);
+            var expected = new Plane(eNormal, eRootPoint);
+            Assert.That(actual, Is.EqualTo(expected));
+        }
+
+        [TestCase("1,1,9;1,2,14;1,3,20;2,1,11;2,2,17;2,3,23;3,1,15;3,2,20;3,3,26", -21.7240278973, -41.0640090729, 7.3269978417, 1)]
+        public void CreateFittedPlane(string sPoints, double eA, double eB, double eC, double eOffset)
+        {
+            //You can see this example at
+            //https://math.stackexchange.com/questions/99299/best-fitting-plane-given-a-set-of-points/1652383#1652383
+
+            var points = sPoints.Split(';').Select(s => Point3D.Parse(s));
+            var actual = Plane.CreateFittedPlaneFrom(points);
+
+            //normalize (eA, eB, eC and eD).
+            //both 2 following equations mean the same plane.
+            // eq1: Ax + By + Cz + D = 0
+            // eq2:-Ax - By - Cz - D = 0
+            var normal = new Vector3D(eA, eB, eC);
+            var eNormalizedOffset = eOffset / normal.Length;
+            var expected = new Plane(normal.Negate().Normalize(), eNormalizedOffset);
+
+            //for debugging purpose
+            TestContext.WriteLine($"{nameof(actual)}={actual}");
+            TestContext.WriteLine($"{nameof(expected)}={expected}");
+
+            var tolerance = 6e-3; // for this example
+            Assert.That(actual.A, Is.EqualTo(expected.A).Within(tolerance), "A");
+            Assert.That(actual.B, Is.EqualTo(expected.B).Within(tolerance), "B");
+            Assert.That(actual.C, Is.EqualTo(expected.C).Within(tolerance), "C");
+            Assert.That(actual.D, Is.EqualTo(expected.D).Within(tolerance), "D");
```

Comment:
`AssertGeometry.AreEqual(expectedPlane, actualPlane);`

Reply:
True. I wish I could have.
I need time to adjust `AssertGeometry.AreEqual(PlaneA, PlaneB)` to pass this test
where
```csharp
PlaneA.Normal == -PlaneB.Normal;
PlaneA.RootPoint == PlaneB.RootPoint;
```

Reply:
This is another problem I spotted in the `Plane` implementation after I took over the maintenance of this repo; the plane is not defined in its [Hesse normal form](https://en.wikipedia.org/wiki/Hesse_normal_form), which I find problematic.

Then the normal direction is uniquely defined and fixed, for any plane. In addition the equality operator should be adjusted, so that we don't check equality of the root points, but that something like `other.Conatins(rootPoint)`, where we check that the root point is contained in the other plane. There is nothing special in the choice of a root point for a plane, any point lying on it is equally valid.

I wanted to open another PR and fix this (breaking) change for the next (v0.8.0) release. Maybe you want to wait with this PR after I am finished (?)